libs/cx/src/lib/nested-list/nested-link/nested-link.component.ts

```diff
@@ -0,0 +1,149 @@
+import {
+    ChangeDetectorRef,
+    ContentChild,
+    Component,
+    ElementRef,
+    EventEmitter,
+    HostBinding,
+    HostListener,
+    Input,
+    Output,
+    Renderer2
+} from '@angular/core';
+import { NestedListButtonDirective, NestedListTitleDirective } from '../nested-list-directives';
+import { NestedItemService } from '../nested-item/nested-item.service';
+import { Nullable } from '@fundamental-ngx/core/shared';
+import { NestedListExpandIconComponent } from '../nested-list-directives';
+
+@Component({
+    template: `
+        <div *ngIf="!_nestedListButton" class="fdx-nested-list__link-container">
+            <ng-container *ngTemplateOutlet="containerContent"></ng-container>
+        </div>
+        <a *ngIf="_nestedListButton" tabindex="0" class="fdx-nested-list__link-container">
+            <ng-container *ngTemplateOutlet="containerContent"></ng-container>
+        </a>
+        <ng-content select="[fdx-nested-list-button]"></ng-content>
+        <ng-template #containerContent>
+            <ng-content select="[fdx-nested-list-icon]"></ng-content>
+            <ng-content select="[fdx-nested-list-title]"></ng-content>
+            <ng-content select="[fdx-nested-list-expand-icon]"></ng-content>
+        </ng-template>
+    `,
+    selector:
+        // eslint-disable-next-line @angular-eslint/component-selector
+        '[fdx-nested-list-link], button[fdx-nested-list-link], a[fdx-nested-list-link], div[fdx-nested-list-link]',
+    host: {
+        '[attr.tabindex]': '!_nestedListButton ? 0 : -1'
+    }
+})
+export class NestedLinkComponent {
+    /** Function that is called on click event dispatch on this element. */
+    @Input()
+    onClickCallback: () => void;
```

Comment:
Why this is not `EventEmitter`? Accepting callback is fine, though it's kinda different principle than emitting events.